controllers/propagator/replicatedpolicy_controller.go

```diff
@@ -0,0 +1,445 @@
+package propagator
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	k8sdepwatches "github.com/stolostron/kubernetes-dependency-watches/client"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1beta1 "open-cluster-management.io/api/cluster/v1beta1"
+	appsv1 "open-cluster-management.io/multicloud-operators-subscription/pkg/apis/apps/placementrule/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	policiesv1 "open-cluster-management.io/governance-policy-propagator/api/v1"
+	"open-cluster-management.io/governance-policy-propagator/controllers/common"
+)
+
+var _ reconcile.Reconciler = &ReplicatedPolicyReconciler{}
+
+type ReplicatedPolicyReconciler struct {
+	Propagator
+}
+
+func (r *ReplicatedPolicyReconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	log := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	log.Info("Reconciling the replicated policy")
+
+	// Set the hub template watch metric after reconcile
+	defer func() {
+		hubTempWatches := r.DynamicWatcher.GetWatchCount()
+		log.V(3).Info("Setting hub template watch metric", "value", hubTempWatches)
+
+		hubTemplateActiveWatchesMetric.Set(float64(hubTempWatches))
+	}()
+
+	replicatedExists := true
+	replicatedPolicy := &policiesv1.Policy{}
+
+	if err := r.Get(ctx, request.NamespacedName, replicatedPolicy); err != nil {
+		if !k8serrors.IsNotFound(err) {
+			log.Error(err, "Failed to get the replicated policy")
+
+			return reconcile.Result{}, err
+		}
+
+		replicatedExists = false
+	}
+
+	rootName, rootNS, err := common.ParseRootPolicyLabel(request.Name)
+	if err != nil && replicatedExists {
+		if err := r.cleanUpReplicated(ctx, replicatedPolicy); err != nil {
+			if !k8serrors.IsNotFound(err) {
+				log.Error(err, "Failed to delete the invalid replicated policy, requeueing")
+
+				return reconcile.Result{}, err
+			}
+		}
+
+		log.Info("Invalid replicated policy deleted")
+
+		return reconcile.Result{}, nil
+	}
+
+	// Fetch the Root Policy instance
+	rootPolicy := &policiesv1.Policy{}
+	rootNN := types.NamespacedName{Namespace: rootNS, Name: rootName}
+
+	if err := r.Get(ctx, rootNN, rootPolicy); err != nil {
+		if k8serrors.IsNotFound(err) {
```

Comment:
Optional: To reduce idents and make the code easier to read, you can do something like:
```go
if !k8serrors.IsNotFound(err) {
		log.Error(err, "Failed to get the root policy, requeueing")

		return reconcile.Result{}, err
}

if !replicatedExists {
	log.Info("Root policy and replicated policy already missing")

	return reconcile.Result{}, nil
}

inClusterNS, err := common.IsInClusterNamespace(r.Client, request.Namespace)
...
```

Reply:
Good idea!